core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.spec.ts

```diff
@@ -1173,8 +1288,72 @@ describe('OptimizedSsrEngine', () => {
         flush();
       }));
     });
-  });
 
+    it('should reuse the same render for subsequent requests for the same rendering key and first request should timeout', fakeAsync(() => {
+      const timeout = 300;
+      const engineRunner = new TestEngineRunner(
+        { timeout, reuseCurrentRendering: true },
+        400,
+        true
+      );
+      jest.spyOn(engineRunner.optimizedSsrEngine as any, 'log');
+      engineRunner.request(requestUrl);
+      tick(200);
+
+      engineRunner.request(requestUrl);
+
+      tick(100);
+      expect(engineRunner.optimizedSsrEngine['log']).toHaveBeenCalledWith(
+        `SSR rendering exceeded timeout ${timeout}, fallbacking to CSR for ${requestUrl}`,
+        false,
+        { request: expect.objectContaining({ originalUrl: requestUrl }) }
+      );
+
+      tick(100);
+      expect(engineRunner.renderCount).toEqual(1);
+      expect(engineRunner.renders).toEqual(['', new Error('a-0')]);
+      flush();
+    }));
+
+    it('should reuse the same render for subsequent requests for the same rendering key when the rendering strategy is ALWAYS_SSR', fakeAsync(() => {
+      const timeout = 300;
+      const engineRunner = new TestEngineRunner(
+        {
+          timeout,
+          reuseCurrentRendering: true,
+          renderingStrategyResolver: () => RenderingStrategy.ALWAYS_SSR,
+        },
+        400,
+        true
+      );
+
+      engineRunner.request(requestUrl);
+      expect(getCurrentConcurrency(engineRunner)).toEqual({
+        currentConcurrency: 1,
+      });
+      expect(getRenderCallbacksCount(engineRunner, requestUrl)).toEqual({
+        renderCallbacksCount: 1,
+      });
+
+      tick(200);
+      engineRunner.request(requestUrl);
+      expect(getCurrentConcurrency(engineRunner)).toEqual({
+        currentConcurrency: 1,
+      });
+      expect(getRenderCallbacksCount(engineRunner, requestUrl)).toEqual({
+        renderCallbacksCount: 2,
+      });
+
+      tick(200);
+
+      expect(engineRunner.renderCount).toEqual(1);
+      expect(engineRunner.renders).toEqual([
+        new Error('a-0'),
+        new Error('a-0'),
+      ]);
+      flush();
+    }));
+  });
```

Comment:
As far as I can see, those 2 tests are related to `reuseCurrentRendering: true` option, but they are placed in the `describe('maxRenderTime option')`. I suggest to move those tests to an appropriate section. IMHO they don't belong to the `maxRenderTime` section.